types/src/account_config/resources/diem_id.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    account_address::AccountAddress, diem_id_identifier::DiemIdVaspDomainIdentifier,
+    event::EventHandle,
+};
+use move_core_types::{
+    ident_str,
+    identifier::IdentStr,
+    move_resource::{MoveResource, MoveStructType},
+};
+use serde::{Deserialize, Serialize};
+
+/// The Identifier for the DiemID module.
+pub const DIEM_ID_MODULE_IDENTIFIER: &IdentStr = ident_str!("DiemId");
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DiemIdDomains {
+    domains: Vec<DiemIdDomain>,
+}
+
+impl DiemIdDomains {
+    pub fn domains(&self) -> &[DiemIdDomain] {
+        &self.domains
+    }
+}
+
+impl MoveStructType for DiemIdDomains {
+    const MODULE_NAME: &'static IdentStr = DIEM_ID_MODULE_IDENTIFIER;
+    const STRUCT_NAME: &'static IdentStr = ident_str!("DiemIdDomains");
+}
+
+impl MoveResource for DiemIdDomains {}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DiemIdDomain {
+    domain: DiemIdVaspDomainIdentifier,
+}
+
+impl DiemIdDomain {
+    pub fn domain(&self) -> &DiemIdVaspDomainIdentifier {
+        &self.domain
+    }
+}
```

Comment:
What's the intent behind this type? wouldn't it just make more sense to just use the `DiemIdVaspDomainIdentifier` directly?

Reply:
Just mirroring the move type structure -- is it ok to not do that and just use DiemIdVaspDomainIdentifier?

Reply:
This just seems like a needless additional wrapper around the thing you really want which is the  `DiemIdVaspDomainIdentifier`. Maybe a Move person has more opinions though